micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/MetricsAspectIntegrationSpec.groovy

```diff
@@ -0,0 +1,143 @@
+package com.ofg.infrastructure.web.resttemplate
+
+import com.codahale.metrics.MetricRegistry
+import com.codahale.metrics.Timer
+import com.ofg.infrastructure.base.MvcWiremockIntegrationSpec
+import com.ofg.infrastructure.web.resttemplate.custom.RestTemplate
+import com.ofg.infrastructure.web.resttemplate.fluent.ServiceRestClientConfiguration
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.SpringApplicationContextLoader
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.EnableAspectJAutoProxy
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpMethod
+import org.springframework.http.RequestEntity
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.web.client.RequestCallback
+import org.springframework.web.client.ResponseExtractor
+import org.springframework.web.client.RestOperations
+
+import javax.annotation.Resource
+import java.util.concurrent.Callable
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*
+
+@ContextConfiguration(classes = [ServiceRestClientConfiguration, Config], loader = SpringApplicationContextLoader)
+class MetricsAspectIntegrationSpec extends MvcWiremockIntegrationSpec {
+
+    @Resource
+    RestOperations restTemplate
+
+    @Autowired
+    MetricRegistry metricRegistry
+
+    URI wireMockUri
+
+    def setup() {
+        wireMockUri = new URI(wireMockUrl)
+    }
+
+    def 'should gather metrics for all RestOperations methods calls'() {
+        given:
+            stubInteraction(any(urlMatching('/.*')), aResponse().withStatus(200))
+
+        expect:
+            verifyMetric('.foo')     { restTemplate.delete("$wireMockUrl/foo") }
+            verifyMetric('.foo._')   { restTemplate.delete("$wireMockUrl/foo/bar") }
+            verifyMetric('.foo._._') { restTemplate.delete("$wireMockUrl/foo/bar/baz") }
+            verifyMetric('.foo._._') { restTemplate.delete("$wireMockUrl/foo/bar/baz") }
+
+            verifyMetric('.id')  { restTemplate.delete("$wireMockUrl/{id}", '42') }
+            verifyMetric('.key') { restTemplate.delete("$wireMockUrl/{key}", [key: 'value']) }
+            verifyMetric         { restTemplate.delete(wireMockUri) }
+
+            def responseType = Void
```

Comment:
Given part in expect block suggests that this should be another (appropriately named) test